### PR TITLE
Chained assertions in JsonPatchProcessorTests class.

### PR DESCRIPTION
**Pull request by CanadianPhilipBarkow**
### Description

Where applicable, assertions have been chained in the `JsonPatchProcessorTests` class. This addresses sonarlint warnings.

### Related Azure Boards Work Items

[AB#4173](https://dev.azure.com/DTS-STN/Canada%20Dental%20Care%20Plan/_workitems/edit/4173)
[AB#4185](https://dev.azure.com/DTS-STN/Canada%20Dental%20Care%20Plan/_workitems/edit/4185)

### Checklist

- [X] I have tested the changes locally
- [X] I have added/updated tests that prove my fix is effective or that my feature works
- [X] I have checked that my code follows the project's coding style by running `npm run format:check`
- [X] I have checked that my code contains no linting errors by running `npm run lint`
- [X] I have checked that my code contains no type errors by running `npm run typecheck`
- [X] I have checked that all unit tests pass by running `npm run test:unit -- run`
- [X] I have checked that all e2e tests pass by running `npm run test:e2e`

### CODE REVIEW

File: backend/src/test/java/ca/gov/dtsstn/cdcp/api/web/json/JsonPatchProcessorTests.java [hunk 1]
```diff
@@ -39,9 +39,7 @@ void testPatchJsonMergePatch() throws BindException{
 		final var jsonMergePatch = Json.createMergePatch(Json.createObjectBuilder(patchObject).build());
 		final var patchedEntity = jsonPatchProcessor.patch(entity, jsonMergePatch);
 
-		assertThat(patchedEntity).isNotSameAs(entity);
-		assertThat(patchedEntity).hasFieldOrPropertyWithValue("id", "id");
-		assertThat(patchedEntity).hasFieldOrPropertyWithValue("name", "updated name");
+		assertThat(patchedEntity).isNotSameAs(entity).hasFieldOrPropertyWithValue("id", "id").hasFieldOrPropertyWithValue("name", "updated name");
```
Comment (gregory-j-baker) [inline backend/src/test/java/ca/gov/dtsstn/cdcp/api/web/json/JsonPatchProcessorTests.java:42]:
```suggestion
		assertThat(patchedEntity)
			.isNotSameAs(entity)
			.hasFieldOrPropertyWithValue("id", "id")
			.hasFieldOrPropertyWithValue("name", "updated name");
```

(for readability)

File: backend/src/test/java/ca/gov/dtsstn/cdcp/api/web/json/JsonPatchProcessorTests.java [hunk 2]
```diff
@@ -72,9 +70,7 @@ void testPatchJsonPatch() throws BindException {
 		final var jsonPatch = Json.createPatch(Json.createArrayBuilder().add(patchObject).build());
 		final var patchedEntity = jsonPatchProcessor.patch(entity, jsonPatch);
 
-		assertThat(patchedEntity).isNotSameAs(entity);
-		assertThat(patchedEntity).hasFieldOrPropertyWithValue("id", "id");
-		assertThat(patchedEntity).hasFieldOrPropertyWithValue("name", "updated name");
+		assertThat(patchedEntity).isNotSameAs(entity).hasFieldOrPropertyWithValue("id", "id").hasFieldOrPropertyWithValue("name", "updated name");
```
Comment (gregory-j-baker) [inline backend/src/test/java/ca/gov/dtsstn/cdcp/api/web/json/JsonPatchProcessorTests.java:73]:
```suggestion
		assertThat(patchedEntity)
			.isNotSameAs(entity)
			.hasFieldOrPropertyWithValue("id", "id")
			.hasFieldOrPropertyWithValue("name", "updated name");
```

Comment (CanadianPhilipBarkow) [reply backend/src/test/java/ca/gov/dtsstn/cdcp/api/web/json/JsonPatchProcessorTests.java]: Sure. Implemented in commit e597aab.

File: backend/src/test/java/ca/gov/dtsstn/cdcp/api/web/json/JsonPatchProcessorTests.java (hunk 1 above)
Comment (CanadianPhilipBarkow) [reply backend/src/test/java/ca/gov/dtsstn/cdcp/api/web/json/JsonPatchProcessorTests.java]: Sure. Implemented in commit e597aab.

**Review by nicholas-ly — APPROVED** [review]

**Pull request merged**